books/calculus/bake

```diff
@@ -0,0 +1,278 @@
+      section.search('.os-problem-container > .os-table:first-child').each do |table|
+        table.add_class('first-element')
+        table.parent.add_class('has-first-element')
+      end
+
+      section.search('.os-problem-container > [data-type="media"]:first-child').each do |img|
+        img.add_class('first-element')
+        img.parent.add_class('has-first-element')
+      end
+    end
+
+    page.search('div.os-table div.os-caption-container').each do |caption|
+      caption_text = caption.first('span.os-caption [data-type="title"]')
+      next unless caption_text.present?
+
+      caption_text.add_class('os-title').name = 'span'
+      divider_aux = caption.first('span.os-divider')
+      divider_aux.append(sibling:
+        <<~HTML
+          #{caption_text&.cut}
+        HTML
+      )
+    end
+  end
+
+  BakeMathInParagraph.v1(book: book)
+  BakeIndex.v1(book: book)
+  BakeCompositePages.v1(book: book)
+  BakeFootnotes.v1(book: book)
+
+  BakeToc.v1(book: book)
+
+  # group composite pages at the end of the chapter
+  book.toc.search('.os-toc-chapter ol.os-chapter').each_with_index do |chapter, index|
+    chapter_element = Nokogiri::XML.fragment(chapter.children)
+    composite_pages = chapter_element.search('li.os-toc-chapter-composite-page')
+    # first_composite_href = composite_pages.search('a').first['href']
+    chapter.search('class').each(&:trash)
+    chapter_element.search('li.os-toc-chapter-composite-page').remove
+    chapter.replace_children(with:
+      <<~HTML
+        #{chapter_element}
+        <li class="os-toc-composite-chapter" cnx-archive-shortid="" cnx-archive-uri="">
+          <a href="#composite-chapter-#{index + 1}">
+            <span class="os-text">#{I18n.t(:chapter_review)}</span>
+          </a>
+          <ol class="os-chapter">
+            #{composite_pages}
+          </ol>
+        </li>
+      HTML
+    )
+  end
+
+  book.chapters.each do |chapter|
+    chapter.search('div[data-type="equation"]:not(.unnumbered)').each do |equation|
+      equation.append(child:
+        <<~HTML
+          <div class="os-equation-number">
+            <span class="os-number">(#{chapter.count_in(:book)}.#{equation.count_in(:chapter)})</span>
+          </div>
+        HTML
+      )
+    end
+
+    BakePageAbstracts.v1(chapter: chapter)
+    chapter.abstracts.each do |abstract|
+      abstract.search('ul').each do |ul|
+        ul.add_class('os-abstract')
+        ul_element = Nokogiri::XML.fragment(ul)
+        ul_element.children.remove
+        ul.search('li').each_with_index do |li, index|
+          ul_element.add_child(
+            <<~HTML
+              <li>
+                <span class="os-abstract-token">#{chapter.count_in(:book)}.#{abstract.count_in(:chapter)}.#{index + 1}</span>
+                <span class="os-abstract-content">#{li.text}</span>
+              </li>
+            HTML
+          )
+        end
+        ul.replace_children(with: ul_element.children)
```

Comment:
Can all of this go into `BakePageAbstracts`?  Check with other people to see if their recipes event have `ul` elements, they may not in which case adding this there may not be a problem.